pink/barriers/body_spherical_barrier.py

```diff
@@ -0,0 +1,189 @@
+            gain: Barrier gain. Defaults to 1.0.
+            safe_displacement_gain: gain for the safe backup displacement
+                cost term. Defaults to 3.0.
+        """
+        if d_min < 0.0:
+            raise NegativeMinimumDistance(
+                "The minimum distance threshold must be non-negative."
+            )
+        super().__init__(
+            dim=1,
+            gain=gain,
+            gain_function=lambda h: h / (1 + np.abs(h)),
+            safe_displacement_gain=safe_displacement_gain,
+        )
+        self.frames = frames
+        self.d_min = d_min
+
+    def compute_barrier(self, configuration: Configuration) -> np.ndarray:
+        r"""Compute the value of the barrier function.
+
+        The barrier function is computed based on the Euclidean distance
+        between the two specified frames. It considers the minimum distance
+        threshold.
+
+        The barrier function is given by:
+
+        .. math::
+
+            h(q) = \|p_1(q) -
+            p_2(q)\|^2 - d_{min}^2
+
+        where :math:`p_1(q)` and
+        :math:`p_2(q)` are the positions of the two
+        frames in the world coordinate system, and :math:`d_{min}`
+        is the minimum distance threshold.
+
+        Args:
+            configuration: Robot configuration :math:`q`.
+
+        Returns:
+            Value of the barrier function
+                :math:`h(q)`.
+        """
+        pos1_world, pos2_world = self._get_frame_positions(configuration)
+        return np.array(
+            [np.linalg.norm(pos1_world - pos2_world) ** 2 - self.d_min**2]
+        )
+
+    def compute_jacobian(self, configuration: Configuration) -> np.ndarray:
+        r"""Compute the Jacobian matrix of the barrier function.
+
+        The Jacobian matrix is computed based on the position Jacobians of the
+        two specified frames. The Jacobians are transformed to align with the
+        world coordinate system.
+
+        The Jacobian matrix is given by:
+
+        .. math::
+
+            \frac{\partial h}{\partial q}(q) =
+            2(p_1(q) -
+            p_2(q))^T \begin{bmatrix}
+                \frac{\partial p_1}{\partial q}
+                (q) \\
+                -\frac{\partial p_2}{\partial q}
+                (q)
+            \end{bmatrix}
+
+        where :math:`\frac{\partial p_1}{\partial q}(q)`
+        and :math:`\frac{\partial p_2}{\partial q}(q)`
+        are the position Jacobians of the two frames.
+
+        Args:
+            configuration: Robot configuration :math:`q`.
+
+        Returns:
+            Jacobian matrix
+                :math:`\frac{\partial h}{\partial q}(q)`.
+        """
+        pos1_world, pos2_world = self._get_frame_positions(configuration)
+        pos1_jac, pos2_jac = self._get_frame_jacobians(configuration)
+
+        dh_dx = 2 * np.concatenate(
+            [pos1_world - pos2_world, pos2_world - pos1_world]
+        )
+        dx_dq = np.vstack([pos1_jac, pos2_jac])
```

Comment:
```suggestion
        dx_dq = np.vstack([pos1_jac, -pos2_jac])
```
If doing a `vstack`, there should be a minus here.